integration_tests/src/lib.rs

```diff
@@ -131,49 +131,50 @@ mod test {
                 read_only_parent: None,
             };
 
-        // XXX Crucible uses std::sync::mpsc::Receiver, not
-        // tokio::sync::mpsc::Receiver, so use tokio::task::block_in_place here.
-        // Remove that when Crucible changes over to the tokio mpsc.
-        let volume = Arc::new(tokio::task::block_in_place(|| {
-            Volume::construct(vcr, None)
-        })?);
+        let volume = Arc::new(Volume::construct(vcr, None).await?);
 
-        volume.activate(0)?;
+        volume.activate(0).await?;
 
         // Verify contents are zero on init
         let buffer = Buffer::new(BLOCK_SIZE * 10);
         volume
-            .read(Block::new(0, BLOCK_SIZE.trailing_zeros()), buffer.clone())?
-            .block_wait()?;
+            .read(Block::new(0, BLOCK_SIZE.trailing_zeros()), buffer.clone())
+            .await?
+            .wait()
+            .await?;
 
-        assert_eq!(vec![0x00_u8; BLOCK_SIZE * 10], *buffer.as_vec());
+        assert_eq!(vec![0x00_u8; BLOCK_SIZE * 10], *buffer.as_vec().await);
 
         // Write data in
         volume
             .write(
                 Block::new(0, BLOCK_SIZE.trailing_zeros()),
                 Bytes::from(vec![0x55; BLOCK_SIZE * 10]),
-            )?
-            .block_wait()?;
+            )
+            .await?
+            .wait()
+            .await?;
 
         // Read parent, verify contents
         let buffer = Buffer::new(BLOCK_SIZE * 10);
         volume
-            .read(Block::new(0, BLOCK_SIZE.trailing_zeros()), buffer.clone())?
-            .block_wait()?;
+            .read(Block::new(0, BLOCK_SIZE.trailing_zeros()), buffer.clone())
+            .await?
+            .wait()
+            .await?;
 
-        assert_eq!(vec![0x55_u8; BLOCK_SIZE * 10], *buffer.as_vec());
+        assert_eq!(vec![0x55_u8; BLOCK_SIZE * 10], *buffer.as_vec().await);
 
         Ok(())
     }
 
-    #[tokio::test(flavor = "multi_thread", worker_threads = 8)]
+    #[tokio::test]
```

Comment:
It's a more constrained environment: by default `tokio::test` runs single-threaded. If we had any locking or awaiting problems it would be exposed by that.